### PR TITLE
[wip]: add the new queries to console code

### PR DESCRIPTION
**Pull request by kolharsam**
So..... what's in this PR?

 - [ ] Hooked up the tables of the event triggers to use the latest APIs
      - [x] Cron
      - [x] One-Off
      - [x] Data (needs fixes on SQL front)
- This PR involves a lot of updates on the FilterQuery component and the `useFilterQuery` method.

#### TODO and What needs fixing
 - The API returns all of the entries within the db and in the case of cron triggers the minimum number of elements on the table just goes up to 60.
- ~~Invocation logs need to be hooked up to a new API so that the details can be fetched properly.~~
(working for cron and one_off)
- ~~The request, response shown in the processed events also need work. (It just crashes now)~~ Working now!

### CODE REVIEW

File: console/src/components/Common/FilterQuery/FilterQuery.tsx [hunk 1]
```diff
@@ -1,41 +1,55 @@
...
 import ReloadEnumValuesButton from '../../Services/Data/Common/Components/ReloadEnumValuesButton';
 import Button from '../Button/Button';
 import { Nullable } from '../utils/tsUtils';
 import styles from './FilterQuery.scss';
 import { BaseTable } from '../../../dataSources/types';
 import { generateTableDef } from '../../../dataSources';
+import { mapDispatchToPropsEmpty } from '../utils/reactUtils';
 
-type Props = {
+interface Props extends InjectedProps {
```
Comment (kolharsam) [inline console/src/components/Common/FilterQuery/FilterQuery.tsx]: @beerose need some help here: the props here are incorrect. This is the error that is reported: ` JSX element type 'FilterQuery' does not have any construct or call signatures.`

File: console/src/components/Common/FilterQuery/FilterQuery.tsx [hunk 2]
```diff
@@ -93,4 +107,29 @@ const FilterQuery: React.FC<Props> = props => {
...
+type PropsFromState = {
+  currentSource?: string;
+  triggerName?: string;
+};
+
+const mapStateToProps: MapStateToProps<
+  PropsFromState,
+  ExternalProps,
+  ReduxState
+> = (state, ownProps) => ({
```
Comment (beerose) [inline console/src/components/Common/FilterQuery/FilterQuery.tsx]:
```suggestion
const mapStateToProps = (state: ReduxState, ownProps: ExternalProps) => ({
```

File: console/src/metadata/queryUtils.ts [hunk 3]
```diff
@@ -629,13 +640,43 @@ export const getConsoleStateQuery = {
...
+      trigger_name,
+    },
+    undefined,
+    true
+  );
+
+export const deleteScheduledEvent = (type: SupportedEvents, event_id: string) =>
+  getMetadataQuery(
+    'delete_scheduled_event',
+    'default',
```
Comment (beerose) [inline console/src/metadata/queryUtils.ts]: we shouldn't pass source at all

File: console/src/metadata/queryUtils.ts [hunk 4]
```diff
@@ -629,13 +640,43 @@ export const getConsoleStateQuery = {
...
+    true
+  );
+
+export const getScheduledEvents = (
+  type: SupportedEvents,
+  trigger_name?: string // TODO. to make this strictly required if type = 'cron'
+) =>
+  getMetadataQuery(
+    'get_scheduled_events',
+    'default',
```
Comment (beerose) [inline console/src/metadata/queryUtils.ts]: we shouldn't pass the source. server will take care of that and will make this query on the proper source

File: console/src/metadata/queryUtils.ts [hunk 5]
```diff
@@ -629,13 +640,43 @@ export const getConsoleStateQuery = {
...
-});
+export type SupportedEvents = 'cron' | 'one_off';
+
+export const getEventInvocationsLogByID = (
+  type: SupportedEvents,
+  event_id: string
+) =>
+  getMetadataQuery(
+    'get_event_invocations',
+    'default',
```
Comment (beerose) [inline console/src/metadata/queryUtils.ts]: we shouldn't pass the source. server will take care of that and will make this query on the proper source

File: console/src/metadata/queryUtils.ts [hunk 6]
```diff
@@ -92,13 +95,21 @@ export const getMetadataQuery = (
...
   args: MetadataQueryArgs;
   version?: number;
 } => {
+  if (noPrefix) {
+    return {
+      type,
+      args: { ...args, source },
+      ...options,
+    };
+  }
```
Comment (beerose) [inline console/src/metadata/queryUtils.ts]: Please remove

File: console/src/metadata/queryUtils.ts [hunk 7]
```diff
@@ -92,13 +95,21 @@ export const getMetadataQuery = (
   type: MetadataQueryType,
   source: string,
   args: MetadataQueryArgs,
-  options?: { version: number }
+  options?: { version: number },
+  noPrefix?: boolean
```
Comment (beerose) [inline console/src/metadata/queryUtils.ts]:
```suggestion
```

File: console/src/components/Services/Events/Container.tsx [hunk 8]
```diff
@@ -79,6 +79,33 @@ const Container: React.FC<Props> = props => {
...
+            value={JSON.stringify([currentDataSource, currentDriver])}
+          >
+            {dataSources.map(s => (
+              <option key={s.name} value={JSON.stringify([s.name, s.driver])}>
+                {s.name} ({s.driver})
+              </option>
+            ))}
+          </select>
+        </Link>
+      </li>
```
Comment (beerose) [inline console/src/components/Services/Events/Container.tsx]:
```suggestion
```

**Review by beerose — CHANGES_REQUESTED** [review]

**Comment by kolharsam** [comment]
@beerose updated the description of the PR for full details on what I'm trying to do in this PR.

File: console/src/components/Services/Events/Common/Components/EventsSubTable.tsx [hunk 9]
```diff
@@ -1,85 +1,183 @@
...
+        error: null,
+      }))
+      .catch(err => ({
+        invocations: [],
+        error: err,
+      })),
+});
+const connector = connect(null, mapDispatchToProps);
+type InjectedReduxProps = ConnectedProps<typeof connector>;
+const connectedEventSubTable = connector(EventsSubTable);
```
Comment (beerose) [inline console/src/components/Services/Events/Common/Components/EventsSubTable.tsx]:
```suggestion
const ConnectedEventSubTable = connector(EventsSubTable);
```

This is a component.

File: console/src/components/Services/Events/Common/Components/EventsSubTable.tsx [hunk 10]
```diff
@@ -1,85 +1,183 @@
...
+    dispatch(getEventInvocationThunk(triggerType, event_id))
+      .then(data => ({
+        invocations: data.invocations,
+        error: null,
+      }))
+      .catch(err => ({
+        invocations: [],
+        error: err,
+      })),
+});
```
Comment (beerose) [inline console/src/components/Services/Events/Common/Components/EventsSubTable.tsx]: Don't use side effects in connect argument functions.

File: console/src/components/Services/Data/DataPageContainer.js [hunk 11]
```diff
@@ -14,7 +14,8 @@ import {
 } from './DataActions';
 import { CLI_CONSOLE_MODE } from '../../../constants';
 import styles from '../../Common/TableCommon/Table.scss';
-import { currentDriver, dataSource, useDataSource } from '../../../dataSources';
+// import { currentDriver, dataSource, useDataSource } from '../../../dataSources';
```
Comment (beerose) [inline console/src/components/Services/Data/DataPageContainer.js]:
```suggestion
```

File: console/src/components/Services/Data/DataSourceContainer.tsx [hunk 12]
```diff
@@ -11,7 +11,7 @@ import { showErrorNotification } from '../Common/Notification';
 import {
   fetchDataInit,
   fetchFunctionInit,
-  updateSchemaInfo,
+  // updateSchemaInfo,
```
Comment (beerose) [inline console/src/components/Services/Data/DataSourceContainer.tsx:14]:
```suggestion
```

**Review by beerose — APPROVED** [review]

**Pull request merged**